ovos_phal_plugin_hotkeys/__init__.py

```diff
@@ -11,20 +11,142 @@
 # limitations under the License.
 #
 import json
+from json.decoder import JSONDecodeError
+from os import walk
+from os.path import exists, join, basename, dirname
+
 from ovos_bus_client.message import Message
 from ovos_plugin_manager.phal import PHALPlugin
 from ovos_utils.log import LOG
+from ovos_utils.xdg_utils import XDG_DATA_DIRS, XDG_DATA_HOME
+
+# TODO: Update ovos-PHAL to differentiate sj201 v6/v10
+# Currently, it only detects v6
+from ovos_PHAL.detection import is_respeaker_2mic, is_mycroft_sj201
+
+from ovos_plugin_manager.templates.phal import PHALValidator
 
 import ovos_phal_plugin_hotkeys.keyboard as keyboard
 
 
+def get_preconfigured_devices():
+    local_config = join(dirname(__file__), "config")
+    device_dir = "OpenVoiceOS/hotkey_devices"
+    devices = []
+    for (p, d, f) in walk(local_config):
+        for filename in f:
+            devices.append(join(p, filename))
+    for directory in XDG_DATA_DIRS:
+        for (p, d, f) in walk(join(directory, device_dir)):
+            for filename in f:
+                devices.append(join(p, filename))
+    for (p, d, f) in walk(join(XDG_DATA_HOME, device_dir)):
+        for filename in f:
+            devices.append(join(p, filename))
+    LOG.debug(f"devices found {devices}")
+    return devices
+
+
+# File is defined in ovos-i2csound
+# https://github.com/OpenVoiceOS/ovos-i2csound/blob/dev/ovos-i2csound#L76
+I2C_PLATFORM_FILE = "/etc/OpenVoiceOS/i2c_platform"
+
+
+class HotKeysPluginValidator(PHALValidator):
+    @staticmethod
+    def validate(config=None):
+        # If the user enabled the plugin no need to go further
+        if config.get("enabled"):
+            LOG.debug("user enabled")
+            return True
+        # This plugin needs configuration to work.  If there is none, no need to load
+        if "key_up" in config or "key_down" in config:
+            # User has predefined settings.
+            LOG.debug("User manually defined a configuration")
+            return True
+        # Check for pre-configured HAT files
+        if get_preconfigured_devices():
+            LOG.debug("Pre-configured HAT's found")
+            return True
```

Comment:
this will always return True, since get_preconfigured_devices includes the bundled json files

Reply:
I forgot to remove that with my changes